doc/proposals/openshift-4.3/operator-testing-tool.md

```diff
@@ -0,0 +1,240 @@
+---
+title: operator-testing-tool
+authors:
+  - "@jmccormick2001"
+  - "@joelanford"
+reviewers:
+  - TBD
+  - "@joelanford"
+approvers:
+  - TBD
+creation-date: 2019-09-13
+last-updated: 2019-09-13
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - "/enhancements/this-other-neat-thing.md"  
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# operator-testing-tool
+
+
+## Release Signoff Checklist
+
+- \[ \] Enhancement is `implementable`
+- \[ \] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created.
+
+## Summary
+
+This proposal is for a new Operator Testing Tool that validates Operator packaging as well as run simple functional test specific to the Operator’s APIs.
+
+## Motivation
+
+Operator Testing and Validation remains difficult today due to tooling sprawl, disparate definitions of validity, and the lack of means to provide custom tests per Operator in a way that is not tied to a particular CI pipeline.
+
+Having high quality Operators is crucial to the success of the Operator Framework.
+
+### Goals
+
+ * Operator Developers can use a tool that reports on recommended, required and optional packaging requirements of a given bundle on disk
+ * Operator Developers can use the same tool that reports on recommended, required and optional common black box tests of a deployed operator on cluster
+ * Operator Developers can rely on a central source of truth of a valid Operator bundles shared among all Operator Framework components
+
+
+### Non-Goals
+
+not in scope: Operator Developers can use the same tool to run custom, functional tests of their Operator on cluster
+
+## Proposal
+
+### User Stories 
+
+#### Story 1 - Show pass/fail in Scorecard Output
+Today, the scorecard output shows a percentage of tests that were successful to the end user. This story is to change the scorecard output to show a pass or fail for each test that is run in the output instead of a success percentage. 
+
+The exit code of scorecard would be 0 if all tests passed. The exit code would be non-zero if tests failed. With this change scores are essentially replaced with a list of pass/fail(s).
+
+The scorecard would by default run all tests regardless if a single test fails.  Using a CLI flag such as below would cause the test execution to stop on a failure:
+ * operator-sdk scorecard -l ‘testgroup=required’ --fail-fast
```

Comment:
Since we will no longer have scores I think we should change the name in order to make it more intuitive. Also, would not it be done outside of the operator-SDK? If yes, I think it should be something as `operator-sdk-qa`

Reply:
I think the name "scorecard" still holds since we're still running a bunch of generic black-box tests and reporting to developers (and eventually users, I think) about which tests passed and failed.